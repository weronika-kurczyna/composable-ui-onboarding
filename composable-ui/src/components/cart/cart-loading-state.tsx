import {
  Divider,
  Skeleton,
  SkeletonText,
  Box,
  BoxProps,
} from '@chakra-ui/react'

interface CartLoadingStateProps {
  rootProps?: BoxProps
}

export const CartLoadingState = ({ rootProps }: CartLoadingStateProps) => {
  return (
    <Box my={{ base: '8', md: '12' }} {...rootProps}>
      <Box display="flex">
        <Skeleton width="200px" height="110px" />
        <Skeleton flex="1" height="110px" ml="20px" />
      </Box>

      <Divider my="40px" />

      <Box display="flex" justifyContent="flex-end">
        <Box>
          <SkeletonText width="360px" />
          <SkeletonText width="360px" my="20px" />
        </Box>
      </Box>
    </Box>
  )
}
