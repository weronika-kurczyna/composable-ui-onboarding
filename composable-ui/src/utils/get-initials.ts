export const getInitials = (str: string) => {
  const names = str.split(' ')
  const initial = names[0].substring(0, 1).toUpperCase()

  return names.length > 1
    ? initial + names[names.length - 1].substring(0, 1).toUpperCase()
    : initial
}
