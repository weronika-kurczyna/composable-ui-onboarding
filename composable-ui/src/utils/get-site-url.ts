export const getSiteUrl = () => {
  // Vercel Support
  if (process.env.NEXT_PUBLIC_VERCEL_URL) {
    return `https://${process.env.NEXT_PUBLIC_VERCEL_URL}`
  }

  // Netlify Support
  if (process.env.NEXT_PUBLIC_NETLIFY_URL) {
    return process.env.NEXT_PUBLIC_NETLIFY_URL
  }

  // Localhost
  return `http://localhost:${process.env.NEXT_PUBLIC_PORT ?? 3000}`
}
