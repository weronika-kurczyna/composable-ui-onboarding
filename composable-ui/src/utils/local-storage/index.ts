// From: https://github.com/rehooks/local-storage
export { writeStorage, deleteFromStorage } from './local-storage-events'
export * from './storage'
export { useLocalStorage } from './use-local-storage'
