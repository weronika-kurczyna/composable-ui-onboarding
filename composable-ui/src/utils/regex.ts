// Define the regular expressions
export const nameRegex = /^[a-zA-Z\u00C0-\u024F\s\-'`]+$/
export const specialCharacterRegex =
  /[\!\@\#\$\%\^\&\*\(\)_\+\-\=\'\:\"\\\|,\.\?]/
export const lowercaseLettersRegex = /[a-z]/
export const uppercaseLettersRegex = /[A-Z]/
export const numbersRegex = /[0-9]/
