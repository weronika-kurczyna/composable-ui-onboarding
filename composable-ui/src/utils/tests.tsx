/* istanbul ignore file */
import React from 'react'
import { IntlProvider as ReactIntlProvider } from 'react-intl'
import {
  render as rtlRender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react'
import { theme } from '@composable/ui'
import { queries, Queries } from '@testing-library/dom'
import { ChakraProvider } from '@chakra-ui/react'

export * from '@testing-library/react'

export function render<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container
>(
  ui: React.ReactElement,
  {
    locale = 'en-US',
    translations = {},
    ...renderOptions
  }: { locale?: string; translations?: Record<string, string> } & RenderOptions<
    Q,
    Container,
    BaseElement
  > = {}
): RenderResult<Q, Container, BaseElement> {
  const Wrapper = () => {
    return (
      <ReactIntlProvider locale="en-US" messages={translations}>
        <ChakraProvider theme={theme}>{ui}</ChakraProvider>
      </ReactIntlProvider>
    )
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}
