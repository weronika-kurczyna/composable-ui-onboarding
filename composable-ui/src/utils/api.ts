/**
 * This is the client-side entrypoint for your tRPC API.
 * It's used to create the `api` object which contains the Next.js App-wrapper
 * as well as your typesafe react-query hooks.
 *
 * We also create a few inference helpers for input and output types
 */
import { httpBatchLink, loggerLink } from '@trpc/client'
import { createTRPCNext } from '@trpc/next'
import { type inferRouterInputs, type inferRouterOutputs } from '@trpc/server'
import superjson from 'superjson'

import { type AppRouter } from '../server/api/root'
import { getSiteUrl } from './get-site-url'

const getBaseUrl = () => {
  if (typeof window !== 'undefined') return '' // browser should use relative url
  return getSiteUrl()
}

async function getCSRFToken() {
  const csrfTokenResponse = await fetch('/api/auth/csrf')
  const { csrfToken } = await csrfTokenResponse.json()
  return csrfToken
}

/**
 * A set of typesafe react-query hooks for your tRPC API
 */
export const api = createTRPCNext<AppRouter>({
  config() {
    return {
      /**
       * React query defaults
       * https://tanstack.com/query/latest/docs/react/guides/important-defaults
       */
      queryClientConfig: {
        defaultOptions: {
          queries: {
            staleTime: 2 * 60 * 1000,
            cacheTime: 5 * 60 * 1000,
            refetchOnWindowFocus: false,
          },
        },
      },

      /**
       * Transformer used for data de-serialization from the server
       * @see https://trpc.io/docs/data-transformers
       **/
      transformer: superjson,

      /**
       * Links used to determine request flow from client to server
       * @see https://trpc.io/docs/links
       * */
      links: [
        loggerLink({
          enabled: (opts) =>
            process.env.NODE_ENV === 'development' ||
            (opts.direction === 'down' && opts.result instanceof Error),
        }),
        httpBatchLink({
          url: `${getBaseUrl()}/api/trpc`,
          async fetch(input, init) {
            if (init && init.method !== 'GET') {
              const token = await getCSRFToken()
              init.headers = { ...init.headers, 'x-csrf-token': token }
            }
            return fetch(input, init)
          },
        }),
      ],
    }
  },
  /**
   * Whether tRPC should await queries when server rendering pages
   * @see https://trpc.io/docs/nextjs#ssr-boolean-default-false
   */
  ssr: false,
})

/**
 * Inference helper for inputs
 * @example type HelloInput = RouterInputs['example']['hello']
 **/
export type RouterInputs = inferRouterInputs<AppRouter>

/**
 * Inference helper for outputs
 * @example type HelloOutput = RouterOutputs['example']['hello']
 **/
export type RouterOutputs = inferRouterOutputs<AppRouter>
