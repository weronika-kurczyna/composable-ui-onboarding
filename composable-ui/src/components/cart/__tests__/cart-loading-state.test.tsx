import { render } from 'utils/tests'
import { CartLoadingState } from '../cart-loading-state'

describe('CartLoadingState', () => {
  it('matches the snapshot', () => {
    const screen = render(<CartLoadingState />)
    expect(screen.container).toMatchSnapshot()
  })
})
