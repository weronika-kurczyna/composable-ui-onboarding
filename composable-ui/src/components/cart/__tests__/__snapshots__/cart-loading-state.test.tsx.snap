// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartLoadingState matches the snapshot 1`] = `
<div>
  <div
    class="css-1olhz9h"
  >
    <div
      class="css-k008qs"
    >
      <div
        class="chakra-skeleton css-1v940jh"
      />
      <div
        class="chakra-skeleton css-1b5pt2q"
      />
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-c4o5e2"
    />
    <div
      class="css-1bvc4cc"
    >
      <div
        class="css-0"
      >
        <div
          class="chakra-skeleton__group css-3mdty7"
        >
          <div
            class="chakra-skeleton css-ctcxab"
          />
          <div
            class="chakra-skeleton css-ctcxab"
          />
          <div
            class="chakra-skeleton css-12r9lpq"
          />
        </div>
        <div
          class="chakra-skeleton__group css-6dcp1f"
        >
          <div
            class="chakra-skeleton css-ctcxab"
          />
          <div
            class="chakra-skeleton css-ctcxab"
          />
          <div
            class="chakra-skeleton css-12r9lpq"
          />
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
